### PR TITLE
Make tests optional and require c++ stdlib 11

### PR DESCRIPTION
**Pull request by MarcusZuber**
Fixes https://github.com/ufo-kit/ufo-core/issues/181

### CODE REVIEW

**Comment by MarcusZuber** [comment]
@tfarago Could you have a look at the meson configs? Could this be done better?

**Comment by tfarago** [comment]
Maybe [compiler-options](https://mesonbuild.com/Builtin-options.html#compiler-options) instead of `override_options`? I didn't even find it in the docs, where is it documented?

**Comment by eli-schwartz** [comment]
https://mesonbuild.com/Reference-manual_functions.html#executable

> override_options: takes an array of strings in the same format as `project`'s `default_options` overriding the values of these options for this target only.

File: tests/unit/meson.build [hunk 1]
```diff
@@ -13,5 +13,6 @@ test('unit tests',
         include_directories: include_dir,
         dependencies: deps,
         link_with: lib,
+        override_options : ['c++_std=c++11']
```
Comment (eli-schwartz) [inline tests/unit/meson.build:16]:
```suggestion
        override_options : ['cpp_std=c++11']
```

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

**Comment by eli-schwartz** [comment]
BTW is there any reason you need to exclude the testsuite? You could just not run it.

Even when used as a subproject, you can use the `--suite` option to `meson test` to only run the tests for the main project, but not subprojects.

File: tests/unit/meson.build (hunk 1 above)
Comment (MarcusZuber) [reply tests/unit/meson.build:16]: Oh, you are right. I will fix this.

**Comment by MarcusZuber** [comment]
> https://mesonbuild.com/Reference-manual_functions.html#executable
> 
> > override_options: takes an array of strings in the same format as `project`'s `default_options` overriding the values of these options for this target only.

Correct. I wanted that this would not interfere with potential other requested versions in other targets.

**Comment by MarcusZuber** [comment]
> BTW is there any reason you need to exclude the testsuite? You could just not run it.
> 
> Even when used as a subproject, you can use the `--suite` option to `meson test` to only run the tests for the main project, but not subprojects.

The idea behind it was, that you can compile the rest, even when meson does not find a c++ compiler implementing the requested version.

I'm not sure if meson just adds only the compiler option or if it checks something. On the machines I had to hand all version where available that meson allows to set, so I could not test it.

**Review by tfarago — APPROVED** [review]

**Pull request merged**